src/evict/evict_lru.c

```diff
@@ -1950,6 +1991,14 @@ __evict_walk_tree(WT_SESSION_IMPL *session, WT_EVICT_QUEUE *queue, u_int max_ent
     ref = btree->evict_ref;
     btree->evict_ref = NULL;
 
+    /*
+     * Get the snapshot for the eviction server when we want to evict dirty content under cache
+     * pressure. This snapshot is used to check for the visibility of the last modified transaction
+     * id on the page.
+     */
+    if (F_ISSET(cache, WT_CACHE_EVICT_DIRTY_HARD | WT_CACHE_EVICT_UPDATES_HARD))
```

Comment:
would it be better to have a bool here, like we do for [eviction workers](https://github.com/wiredtiger/wiredtiger/blob/b53fc8fcdf4ee2abec8bc3b1153097bed4e697d8/src/evict/evict_page.c#L850). Then we check against that bool every time, instead of `F_ISSET(cache, WT_CACHE_EVICT_DIRTY_HARD | WT_CACHE_EVICT_UPDATES_HARD) `or `if (F_ISSET(txn, WT_TXN_HAS_SNAPSHOT))`

Reply:
We don't need to distinguish application thread and eviction thread here. so F_ISSET(txn, WT_TXN_HAS_SNAPSHOT) is enough

Reply:
fair, even [here](https://github.com/wiredtiger/wiredtiger/pull/10576/files#diff-fba173dbad7997f847117319e1939715f1969181ab2937508a2c9303b201f017R1806) we should have only HAS_SNAPSHOT then?

Reply:
I don't want to assume that if we have a snapshot, we are under cache pressure so I check both.

Reply:
I am more inclined towards an assertion but don't mind this.